src/plugins/controls/public/react_controls/controls/data_controls/range_slider/get_range_slider_control_factory.tsx

```diff
@@ -113,14 +111,15 @@ export const getRangesliderControlFactory = (
 
       const dataLoadingSubscription = combineLatest([loadingMinMax$, loadingHasNoResults$])
         .pipe(
+          debounceTime(100),
```

Comment:
I added a small debounce here to prevent the loading state from "flashing" as often as it did before.